cactus/tests/test_plugins.py

```diff
@@ -71,4 +76,143 @@ def test_call(self):
 
         #postBuild
         self.assertEqual(1, len(plugin.postBuild.calls))
-        self.assertEqual((self.site,), plugin.postBuild.calls[0]['args'])
\ No newline at end of file
+        self.assertEqual((self.site,), plugin.postBuild.calls[0]['args'])
+
+
+class TestPagecontextPlugin(unittest.TestCase):
+
+    def test_jekyll_style(self):
+        testdata = [
+            (['---',  # fenced
+              'Author : Me',
+              'Content: Awesome',
+              '---',
+              'Here comes the content'],
+             {'Author': 'Me', 'Content': 'Awesome'},
+             ['Here comes the content']),
+
+            (['Author : Me',  # no start of metadata
+              '---',
+              'Here comes the content'],
+             {},
+             ['Author : Me',
+              '---',
+              'Here comes the content']),
+
+            (['',  # file starts with a blank line
+              '---',
+              'Todays motto: whatever'],
+             {},  # no context data
+             ['', '---', 'Todays motto: whatever']),  # everything preserved
+
+            (['This file has',
+              'no metadata whatsoever'],
+             {},
+             ['This file has', 'no metadata whatsoever']),
+
+        ]
+        plugin = PageContextPlugin()
+
+        for i, (lines, context, data) in enumerate(testdata):
```

Comment:
Can you rename those variables a bit to make the intent clearer?

`lines` could be `input_lines`
`context` could be `expected_context`
`data` could be `expected_output_lines`

Or something like this. This was a bit difficult to understand when I read it for the first time.